10up/block-catalog
# Add workflow to run PHP compat testing

### Description of the Change

Adds our (semi) standard workflow to run PHP compatibility scanning. While this can run as part of the PHPCS scanning, this PR removes the PHPCompatibilityWP standard from our normal PHPCS scanning and adds a new workflow just for that. Open to opinions on if these should be separated or not, the one benefit is it makes it clear when there's a normal PHPCS issue and when there's a compatibility issue.

This compat Action does the following:

- Runs anytime a PR is opened against `develop` or code is pushed to `develop` or `trunk`
- Only runs if PHP files have been changed
- Sets up PHP version 7.4 (our minimum), runs `composer install` and then lints any of the changed PHP files, using our php-compat.xml standard

Closes #16

### How to test the Change

Can see a commit with no PHP changes here: https://github.com/10up/block-catalog/pull/22/checks?sha=010716195d0bb33168772f5c709d40035263b288

Can see a commit with a failing PHP compat check but successful PHPCS check here: https://github.com/10up/block-catalog/pull/22/checks?sha=1ab8288f9996fc7e104876f272d7518f835445d3

### Changelog Entry

> Added - Workflow to run PHP compatibility scanning.

### Credits

Props @dkotter

### Checklist:
